packages/smooth_app/lib/database/search_history.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/foundation.dart';
+import 'package:sqflite/sqflite.dart';
+
+class SearchHistory extends ChangeNotifier {
+  SearchHistory(this.db);
+
+  static const int maxSize = 5;
+
+  final Database db;
+  static const String _tableName = 'search_history';
+  static const String _columnId = 'id';
+  static const String _columnQuery = 'query';
+
+  static Future<void> onUpgrade(
+    final Database db,
+    final int oldVersion,
+    final int newVersion,
+  ) async {
+    if (oldVersion < 9) {
+      return db.execute('''
+        CREATE TABLE $_tableName(
```

Comment:
I would suggest something like that:
```sql
create table $_tableName(
   query TEXT primary key,
   ts TIMESTAMP not null
)
```
"My" `ts` column gives an additional info, compared to "your" `id` column.
Besides, the `query` column being not null and unique looks a lot like a primary key, doesn't it?

Reply:
@slava-sh Technically it's not the ROWID that you're using: it's an additional auto-incremented explicit int ID, with ascending values. You use two unique non null indexes, I use one. Well, it's only a small table anyway...

Reply:
According to the [documentation](https://www.sqlite.org/lang_createtable.html#rowids_and_the_integer_primary_key), the `id` column is an alias for the `rowid`, not a separate column:

> With one exception noted below, if a rowid table has a primary key that consists of a single column and the declared type of that column is "INTEGER" in any mixture of upper and lower case, then the column becomes an alias for the rowid. Such a column is usually referred to as an "integer primary key". A PRIMARY KEY column only becomes an integer primary key if the declared type name is exactly "INTEGER". Other integer type names like "INT" or "BIGINT" or "SHORT INTEGER" or "UNSIGNED INTEGER" causes the primary key column to behave as an ordinary table column with integer affinity and a unique index, not as an alias for the rowid.
>
> The exception mentioned above is that if the declaration of a column with declared type "INTEGER" includes an "PRIMARY KEY DESC" clause, it does not become an alias for the rowid and is not classified as an integer primary key...

Reply:
Fair enough.
Having to read the fine print is not something I would encourage though - especially if you have to read other fine prints to check if the rowids are given ascending values. Which is the case in 99.999% of the time.
That's why for a shared code I still believe my table suggestion was more readable and appropriate.

Anyway given the probable database refactoring I would even recommend storing something more simple, like just a list of barcodes, in json, stored in whatever database system we'll use in the future. We'll see that later.